clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -219,7 +219,7 @@ private long listOffset(TopicPartition partition, long timestamp) {
             for (PartitionRecords<K, V> part : this.records) {
                 Long consumed = subscriptions.consumed(part.partition);
                 if (this.subscriptions.assignedPartitions().contains(part.partition)
-                    && (consumed == null || part.fetchOffset == consumed)) {
```

Comment:
Since we already do the checking while handling the fetch response, logically when we call fetchedRecords this case (consumed == null || part.fetchOffset == consumed) should not happen, right? In this case shall we return an IllegalStateException if this really happens here?